### PR TITLE
Making gradient clipping optional & max gradient norm variable

### PR DESCRIPTION
**Pull request by plonerma**
This PR allows users to disable gradient clipping (by setting `max_grad_norm=None`) and makes the maximum norm a gradient is clipped to variable.

### CODE REVIEW

**Comment by alanakbik** [comment]
@plonerma thanks for adding this!
